File: concept/17_recursion/power/power.dart
import 'dart:io';

void main() {
  print("Enter number : ");
  int input = int.parse(stdin.readLineSync()!);

  int result = powerOf(input);

  print("Number is $input result is $result");
}

int powerOf(int n) {
  if (n == 0) {
    return 1;
  }

  int small = powerOf(n - 1);
  int result = 2 * small;

  return result;
}

File: concept/17_recursion/power/power.cpp
#include <iostream>
using namespace std;

int powerOf(int m)
{
    if (m == 0)
    {
        return 1;
    }

    int small = powerOf(m - 1);
    int big = 2 * small;
    return big;
}
int main()
{
    int n;
    cout << "Enter number : ";
    cin >> n;

    int result = powerOf(n);
    cout << "Power of " << n << " is " << result << endl;
}